src/ixdat/techniques/deconvolution.py

```diff
@@ -0,0 +1,254 @@
+            """
+
+
+        t_sig, v_sig = self.get_calib_signal(signal_name,
+            tspan=tspan,
+            t_bg=t_bg)
+
+        kernel = kernel_obj.generate_kernel(dt=t_sig[1]-t_sig[0])
+        kernel = np.hstack((kernel, np.zeros(len(sig) - len(kernel))))
+        H = fft(kernel)
+
+        partial_current = np.real(ifft(
+            fft(v_sig)*np.conj(H)/(H*np.conj(H) + (1/snr)**2)
+            ))
+
+        return t_sig, partial_current
+
+
+    def extract_kernel(self, signal_name, cutoff_pot=0, tspan=None, t_bg=None):
+        """Extracts a Kernel object from a measurement.
+
+        Args:
+            signal_name (str): Signal name from which the kernel/impule
+                response is to be extracted.
+            cutoff_pot (int): Potential which the defines the onset of the
+                impulse. Must be larger than the resting potential before the
+                impulse.
+            tspan(list): Timespan from which the kernel/impulse response is
+                extracted.
+            t_bg (list): Timespan that corresponds to the background signal.
+        """
+        x_curr, y_curr = self.get_current(tspan=tspan)
+        x_pot, y_pot = self.get_potential(tspan=tspan)
+        x_sig, y_sig = self.get_signal(signal_name, tspan=tspan, t_bg=t_bg)
+
+        if signal_name == 'M32':
+            t0 = x_curr[np.argmax(y_pot > cutoff_pot)] #time of impulse
+        elif signal_name == 'M2':
+            t0 = x_curr[np.argmax(y_pot < cutoff_pot)]
+        else:
+            print('mass not found')
+
+        x_sig = x_sig - t0
+
+        y_sig = y_sig[x_sig>0]
+        x_sig = x_sig[x_sig>0]
+
+        y_curr = y_curr[x_curr>t0]
+        x_curr = x_curr[x_curr>t0]
+        y_pot = y_pot[x_pot>t0]
+        x_pot = x_pot[x_pot>t0]
+
+        kernel = Kernel(
+            MS_data = np.array([x_sig, y_sig]),
+            EC_data = np.array([x_curr, y_curr, x_pot, y_pot])
+            )
+
+
+        return kernel
+
+
+
+class Kernel:
+    """
+    Kernel class implementing different functionalities for datatreatment of
+    kernel/impulse response data.
+    """
+    def __init__ (
+        self,
+        parameters = {},
+        MS_data = None,
+        EC_data = None,
+    ):
+        """Initializes a Kernel object either in functional form by defining the
+        mass transport parameters or in the measured form by passing of EC-MS
+        data.
+
+        Args:
+            parameters (dict): Dictionary containing the mass transport
+                parameters with the following keys:
+                    D: Diffusion constant in liquid
+                    L: Working distance between electrode and gas/liq interface
+                    V: Gas sampling volume of the chip
+                    V_dot: Volumetric capillary flow
+                    kH: Dimensionless Henry volatility
+                MS_data (list): List of numpy arrays containing the MS signal
```

Comment:
avoid capitalizing variable names other than classes and constants (see [PEP 8](https://www.python.org/dev/peps/pep-0008/)). And why not just let the caller of `Kernel.__init__` just tell Kernel which `Measurement` is in use, and the names of the `VSeries` to use?

Reply:
Agree, on the TODO list for the new Kernel class (this was just copied pasted from the EC_MS deconvolution module).